src/elements/name_section.rs

```diff
@@ -10,92 +10,156 @@ const NAME_TYPE_LOCAL: u8 = 2;
 
 /// Debug name information.
 #[derive(Clone, Debug, PartialEq)]
-pub enum NameSection {
-	/// Module name section.
-	Module(ModuleNameSection),
-
-	/// Function name section.
-	Function(FunctionNameSection),
-
-	/// Local name section.
-	Local(LocalNameSection),
-
-	/// Name section is unparsed.
-	Unparsed {
-		/// The numeric identifier for this name section type.
-		name_type: u8,
-		/// The contents of this name section, unparsed.
-		name_payload: Vec<u8>,
-	},
+pub struct NameSection {
+	/// Module name subsection.
+	module_name_subsection: Option<ModuleNameSubsection>,
+
+	/// Function name subsection.
+	function_name_subsection: Option<FunctionNameSubsection>,
```

Comment:
naming might be shortened to "module", "function", "local" as well?

Reply:
Sounds good, but maybe the second and the third in plural ("functions", "locals")?

Reply:
> Sounds good, but maybe the second and the third in plural ("functions", "locals")?

Sure!

Reply:
done